lib/srv/db/sqlserver/proxy.go

```diff
@@ -41,7 +41,7 @@ type Proxy struct {
 
 // HandleConnection accepts connection from a SQL Server client, authenticates
 // it and proxies it to an appropriate database service.
-func (p *Proxy) HandleConnection(ctx context.Context, proxyCtx *common.ProxyContext, tlsConn *tls.Conn) error {
+func (p *Proxy) HandleConnection(ctx context.Context, proxyCtx *common.ProxyContext, tlsConn utils.TLSConn) error {
```

Comment:
I don't see where the `tls.Conn`  is required for below logic we can just use net.Conn here.